models/src/main/scala/coop/rchain/models/rholangN/Basic.scala

```diff
@@ -0,0 +1,168 @@
+package coop.rchain.models.rholangN
+
+final class NilN() extends BasicN
+object NilN { def apply(): NilN = new NilN }
+
+/** *
+  * Rholang process
+  *
+  * For example, `@0!(1) | @2!(3) | for(x <- @0) { Nil }` has two sends
+  * and one receive.
+  */
+final class ParProcN(val ps: Seq[ParN]) extends BasicN {
+  def sortedPs: Seq[ParN]       = ParManager.Manager.sortPars(ps)
+  def addPar(p: ParN): ParProcN = ParProcN(ps :+ p)
+}
+object ParProcN {
+  def apply(ps: Seq[ParN] = Seq()): ParProcN = new ParProcN(ps)
+  def apply(p: ParN): ParProcN               = apply(Seq(p))
+  def empty(): ParProcN                      = ParProcN()
```

Comment:
`apply` is the same as `empty` here. I think better remove default argument for apply method if you have `empty` constructor